zrml/neo-swaps/src/types/max_assets.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2023-2024 Forecasting Technologies LTD.
+//
+// This file is part of Zeitgeist.
+//
+// Zeitgeist is free software: you can redistribute it and/or modify it
+// under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at
+// your option) any later version.
+//
+// Zeitgeist is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+use crate::consts::MAX_ASSETS;
+use sp_runtime::traits::Get;
+
+pub(crate) struct MaxAssets;
+
+impl Get<u32> for MaxAssets {
+    fn get() -> u32 {
+        MAX_ASSETS
+    }
+}
```

Comment:
Why is that necessary? Why don't we use a `Config` type for that?

Reply:
I was worried about a misconfiguration. The thing is that there's a (hard-coded!) minimum spot price required when creating a pool, so there should be a maximum number of assets allowed, because if there are too many assets, it becomes impossible to keep all spot prices above the minimum spot price. I'll add a comment to the code.

Reply:
Seems a bit overkill to me to create a whole new file for that. I was thinking along the lines of putting the following line next to the constant:
```rust
pub type MaxAssets = frame_support::pallet_prelude::ConstU32<MAX_ASSETS>;
```